packages/medusa/src/api/routes/store/carts/index.ts

```diff
@@ -154,6 +154,52 @@ export const defaultStoreCartRelations = [
 /**
  * @schema StoreCartsRes
  * type: object
+ * x-expanded-relations:
+ *   field: cart
+ *   relations:
+ *     - billing_address
+ *     - discounts
+ *     - discounts.rule
+ *     - gift_cards
+ *     - items
+ *     - items.adjustments
+ *     - items.variant
+ *     - payment
+ *     - payment_sessions
+ *     - region
+ *     - region.countries
+ *     - region.payment_providers
+ *     - shipping_address
+ *     - shipping_methods
+ *     - shipping_methods.shipping_option
+ *   implicit:
```

Comment:
question: How do you infer implicit relations?

Reply:
I used `implicit:` to explicitly declare relations that are loaded without the controller's knowledge.

In the case of `StoreCartRes`, we have "eager" loaded relations that are declared on TypeORM models, which I believe we want to move away from in favour of loading them using our expanded relation mechanism. For example, it is the case for `region.payment_providers`. 
https://github.com/medusajs/medusa/blob/831604a5b93797eedd00e8c85a3bd192e597edc2/packages/medusa/src/models/region.ts#L63-L66

The other use case here is to explicitly declare relations that are loaded as part of a subprocess but are not "cleaned" or removed before sending the response. For example, it is the case for `region.tax_rates` that is loaded during the `decorateTotals` process but will not be removed by `defaultStoreCartFields`.
https://github.com/medusajs/medusa/blob/831604a5b93797eedd00e8c85a3bd192e597edc2/packages/medusa/src/services/cart.ts#L2786

In summary, I expect `implicit` to be a temporary measure until implicit relations become explicitly declared in "default" expanded relations or are removed from the response with "default" fields.

Reply:
Got it - thanks for explaining. Makes a lot of sense. And yes, we are moving away from eager-loaded relations in the near future 👍